server/src/subscriber/subscriber.service.ts

```diff
@@ -145,6 +145,39 @@ export class SubscriberService {
     }
   }
 
+  /**
+   * Fetch the user's list of subscriptions.
+   * @param {string} id - The user's zap_production_id or zap_staging_id.
+   * @returns {object}
+   */
+  async getSubscriptions(id: string) {
+    const query = `zap_production_id LIKE '${id}' OR zap_staging_id LIKE '${id}'`
```

Comment:
Let's look for the custom field for the current environment explicitly, instead of hard coding the field names and doing an `OR`. Could you use that sendgrid environment env variable you have in your other branch?